### ISSUE TITLE
Trouble using index backends besides sqlite

### ISSUE THREAD

**Issue by huangsam** [docker/docker-registry#543]
This works on my copy of the registry:

```
docker run \
         -e SEARCH_BACKEND=sqlalchemy \
         -e SQLALCHEMY_INDEX_DATABASE=sqlite:////tmp/docker-registry-index.db \
         -p 5000:5000 \
         registry
```

I create a Docker `mysql` container with the following command:

`docker run --name some-mysql -e MYSQL_ROOT_PASSWORD=mysecretpassword -d -p 3306:3306 mysql`

And that was a success. I then check the container's IP address:

`$ docker inspect -f "{{.NetworkSettings.IPAddress}}" some-mysql`

It was 172.17.0.46. So I went ahead and tried the following:

```
docker run \
         -e SEARCH_BACKEND=sqlalchemy \
         -e SQLALCHEMY_INDEX_DATABASE=mysql://root:mysecretpassword@172.17.0.46:3306/index \
         -p 5000:5000 \
         registry
```

I had the following log messages pop up:

```
2014-08-29 07:42:03,977 WARNING: Cache storage disabled!
2014-08-29 07:42:03,977 WARNING: LRU cache disabled!
2014-08-29 07:42:03,981 DEBUG: Will return docker-registry.drivers.file.Storage
** [Bugsnag] No API key configured, couldn't notify
Traceback (most recent call last):
  File "/usr/local/bin/docker-registry", line 9, in <module>
    load_entry_point('docker-registry==0.8.0', 'console_scripts', 'docker-registry')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 351, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2363, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2088, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "/usr/local/lib/python2.7/dist-packages/docker_registry/run.py", line 23, in <module>
    from .search import *  # noqa
  File "/usr/local/lib/python2.7/dist-packages/docker_registry/search.py", line 17, in <module>
    INDEX = index.load(cfg.search_backend.lower())
  File "/usr/local/lib/python2.7/dist-packages/docker_registry/lib/index/__init__.py", line 82, in load
    return db.SQLAlchemyIndex()
  File "/usr/local/lib/python2.7/dist-packages/docker_registry/lib/index/db.py", line 57, in __init__
    self._engine = sqlalchemy.create_engine(database)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/__init__.py", line 344, in create_engine
    return strategy.create(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/strategies.py", line 73, in create
    dbapi = dialect_cls.dbapi(**dbapi_args)
  File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/connectors/mysqldb.py", line 63, in dbapi
    return __import__('MySQLdb')
ImportError: No module named MySQLdb
```

I then went to [this link](http://stackoverflow.com/questions/454854/no-module-named-mysqldb) to try resolving the problem, but it looks like MySQL configuration was not found.

In short, the `SQLALCHEMY_INDEX_DATABASE` option did not work for me when trying out MySQL. Wondering if this was just intended for just SQLite, or if MySQL has to be co-located in the same container as docker-registry.
